### PR TITLE
Add GHC 9.8 to CI

### PR DESCRIPTION
**Pull request by martijnbastiaan**
## Still TODO:

  - [X] ~~Write a changelog entry (see changelog/README.md)~~
  - [X] ~~Check copyright notices are up to date in edited files~~
  - [X] ~~Add issue mentioning we need to add GHC 9.8 to GitHub Actions (once Stackage has a stable release with 9.8?)~~ Fixed, thanks @christiaanb!

[comment]: # (Depending on the change, some of these points may not be necessary. If so, leave the boxes unchecked so it is easier for a reviewer to see what has been omitted.)

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -66,13 +66,20 @@ optional-packages:
 
 allow-newer:
   brick:base,
+  brick:deepseq,
```
Comment (DigitalBrains1) [inline cabal.project:69]: I think we can drop the following ones:
```
vector-binary-instances:base,
cryptohash-sha256:base,
haskell-src-meta:template-haskell,
string-interpolate:template-haskell,
string-interpolate:text,
tasty-hedgehog:tasty,
```

Comment (DigitalBrains1) [reply cabal.project:69]: Perhaps that also goes for `clash-ffi/example/cabal.project`? I did not test it.

Comment (christiaanb) [reply cabal.project:69]: Thanks, I've made the allow-newer lists smaller.

**Review by DigitalBrains1 — APPROVED** [review]
LGTM, and I'm fine with merging as-is, but I wonder if it should also include dropping the following env var:

https://github.com/clash-lang/clash-compiler/blob/aba55fed9f45711c8336935721a43d243f7f78c1/tests/Main.hs#L66-L70

Because apparently it does not matter that `~/.cabal/store/ghc-9.8.2` does not exist.

Hmmm, wait, I now see that it also ends in `:`! This means that all standard paths are appended. So maybe we shouldn't _drop_ the env var, but modify it to only include `dist-newstyle/packagedb/ghc-9.8.2` which does in fact exist; it turns out the GHC ABI is not appended in that dir (at least not for me, locally).

**Comment by martijnbastiaan** [comment]
Wow, that's weird @DigitalBrains1. I think we should make an issue of it.

I'm happy merging too. Thanks for all the work @christiaanb!

**Pull request merged**